src/sentry/models/environment.py

```diff
@@ -48,8 +48,28 @@ def get_cache_key(cls, organization_id, name):
         return 'env:2:%s:%s' % (organization_id, md5_text(name).hexdigest())
 
     @classmethod
+    def get_name_or_default(cls, name):
+        return name or ''
+
+    @classmethod
+    def get_for_organization_id(cls, organization_id, name):
+        name = cls.get_name_or_default(name)
+
+        cache_key = cls.get_cache_key(organization_id, name)
+
+        env = cache.get(cache_key)
+        if env is None:
+            env = cls.objects.get(
+                name=name,
+                organization_id=organization_id,
+            )
+            cache.set(cache_key, env, 3600)
+
+        return env
+
+    @classmethod
     def get_or_create(cls, project, name):
-        name = name or ''
+        name = cls.get_name_or_default(name)
```

Comment:
❤️ may want to go ahead and use this here? https://github.com/getsentry/sentry/blob/master/src/sentry/tasks/unmerge.py#L321

Reply:
You've uncovered my not so secret plan to inflate my contribution graph numbers